src/Shared/VolumeNameGenerator.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Utils;
+
+internal static class VolumeNameGenerator
+{
+    /// <summary>
+    /// Creates a volume name with the form <c>$"{applicationName}-{resourceName}-{suffix}</c>, e.g. <c>"myapplication-postgres-data"</c>.
+    /// </summary>
+    /// <remarks>
+    /// If the application name contains chars that are invalid for a volume name, the prefix <c>"volume-"</c> will be used instead.
+    /// </remarks>
+    public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suffix) where T : IResource
```

Comment:
super duper nit: should this be `VolumeNameGenerator.Create<T>`?

Reply:
Perhaps? We're not super consistent with naming these kinds of things from what I can see, e.g. `PathNormalizer.NormalizePathForCurrentPlatform` vs. `PasswordGenerator.Generate`.